tests/components/camera/test_local_file.py

```diff
@@ -115,3 +118,38 @@ def test_camera_content_type(hass, aiohttp_client):
     assert resp_4.content_type == 'image/jpeg'
     body = yield from resp_4.text()
     assert body == image
+
+
+@asyncio.coroutine
```

Comment:
Maybe take the opportunity to use they new asyncio syntax

Reply:
Sure, but should that be a seperate PR?

Reply:
@OttoWinter has already stressed his opinion, but since it's just a syntax change I would do it in this PR... 

But again, I'm no official reviewer 😎

Reply:
I mean the above thing with `async_register` **is** a bug (calling async methods from a sync context can lead to some very weird issues).

For this PR: I think the **new** methods added in this PR can/should be converted to `async def` instead of the old `@asyncio.coroutine`. But converting the whole platform is out of the scope of this PR IMO.

Reply:
OK I will have a go. Il be honest I'm not experienced with asyncio but its high time I learn!

Reply:
Basically it's not much more than 

```python
@asyncio.coroutine
def foo():
   yield from bar()
```

becomes:

```python
async def foo():
    await bar()
```

It's mainly a syntax change with the introduction of the new `async` keyword and the replacement of `yield from` with `await` in coroutines.